src/zenml/integrations/databricks/orchestrators/databricks_orchestrator.py

```diff
@@ -0,0 +1,433 @@
+        container_op by pointing at the downstream steps.
+
+        This callable is then compiled into a databricks yaml file that is used as
+        the intermediary representation of the kubeflow pipeline.
+
+        This file, together with some metadata, runtime configurations is
+        then uploaded into the kubeflow pipelines cluster for execution.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+
+        Raises:
+            RuntimeError: If trying to run a pipeline in a notebook
+                environment.
+        """
+        # First check whether the code running in a notebook
+        if Environment.in_notebook():
+            raise RuntimeError(
+                "The Kubeflow orchestrator cannot run pipelines in a notebook "
+                "environment. The reason is that it is non-trivial to create "
+                "a Docker image of a notebook. Please consider refactoring "
+                "your notebook cells into separate scripts in a Python module "
+                "and run the code outside of a notebook when using this "
+                "orchestrator."
+            )
+
+        # Create a callable for future compilation into a dsl.Pipeline.
+        def _construct_databricks_pipeline(
+            zenml_project_wheel: str, cluster_id: str
+        ) -> List[DatabricksTask]:
+            """Create a databrcks task for each step.
+
+            This should contain the name of the step or task and configures the
+            entrypoint of the task to run the step.
+
+            Additionally, this gives each task information about its
+            direct downstream steps.
+            """
+            tasks = []
+            for step_name, step in deployment.step_configurations.items():
+                # image = self.get_image(
+                #    deployment=deployment, step_name=step_name
+                # )
+
+                # The arguments are passed to configure the entrypoint of the
+                # docker container when the step is called.
+                arguments = (
+                    StepEntrypointConfiguration.get_entrypoint_arguments(
+                        step_name=step_name, deployment_id=deployment.id
+                    )
+                )
+
+                # Construct the --env argument
+                env_vars = []
+                for key, value in environment.items():
+                    env_vars.append(f"{key}={value}")
+                env_vars.append("ZENML_DATABRICKS_SOURCE_PREFIX=zenmlproject")
+                env_vars.append(
+                    f"ZENML_DATABRICKS_ORCHESTRATOR_RUN_ID={deployment.id}"
+                )
+                env_arg = ",".join(env_vars)
+
+                arguments.extend(["--env", env_arg])
```

Comment:
Is there no way to pass environment variables to a databricks job? This seems very ugly, and potentially runs into issues with a too long argument string.